packages/xo-web/src/common/store/reducer.js

```diff
@@ -125,7 +128,11 @@ export default {
           }
         }
 
-        return { all, byType, fetched: true }
+        if (initialFetch) {
+          fetched = true
+        }
+
+        return { all, byType, fetched }
```

Comment:
Don't you think a separate action/reducer would simplify the implementation?

Reply:
Yes, it'll facilitate the future debug and we'll reduce the code complexity.